sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -1803,7 +1811,21 @@ private async Task StartPublishingAsync(CancellationToken cancellationToken)
 
                 if ((_partitions == null) || (_partitionHash == null))
                 {
-                    await UpdatePartitionInformation(cancellationToken).ConfigureAwait(false);
+                    // The retry policy for the buffered producer is, by default, generous to allow for long running background work to be done
+                    // effectively. For startup, a more conservative retry policy is needed to prevent the producer from hanging, so cancel
+                    // after one TryTimeout interval.
+
+                    using var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                    linkedCts.CancelAfter(_options.RetryOptions.TryTimeout);
+
+                    try
+                    {
+                        await UpdatePartitionInformation(linkedCts.Token).ConfigureAwait(false);
+                    }
+                    catch (OperationCanceledException)
+                    {
+                        throw new EventHubsException(true, EventHubName, "The buffered producer took too long to start.", EventHubsException.FailureReason.ServiceTimeout);
```

Comment:
Just realized - we should be using the [resources dictionary](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Resources.resx) for the message.   Once we do that, I think we're good to go!